app/src/main/java/com/example/android/navigation/GameWonFragment.kt

```diff
@@ -37,10 +38,36 @@ class GameWonFragment : Fragment() {
             view?.findNavController()?.navigate(GameWonFragmentDirections.actionGameWonFragmentToGameFragment())
         }
         displayArgs()
+        setHasOptionsMenu(true)
         return binding.root
     }
-    private fun displayArgs(){
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.winner_menu, menu)
+        if (null == getShareIntent().resolveActivity(activity!!.packageManager)) {
```

Comment:
Shouldn't we check here if something is null? Seems pretty strange to read if null si equal with smth